xgcm/test/test_grid_ufunc.py

```diff
@@ -81,6 +83,100 @@ def test_invalid_signatures(self, signature):
             _parse_grid_ufunc_signature(signature)
 
 
+class TestGetSignatureFromTypeHints:
+    def test_no_args_to_annotate(self):
+        @as_grid_ufunc()
+        def ufunc():
+            ...
+
+        assert ufunc.signature == "()->()"
+
+    def test_annotated_args(self):
+        # TODO test hints without annotations
+        # TODO test hints with annotations that don't conform to Xgcm
+
+        @as_grid_ufunc()
+        def ufunc(a: Gridded[xr.DataArray, "X:center"]):
```

Comment:
I need to try and change this so that we [type hint with numpy arrays instead](https://github.com/xgcm/xgcm/issues/344#issuecomment-1020344225), i.e.
```python
Gridded[np.ndarray, "X:center"]
```
(or maybe even `ArrayLike` which I saw somewhere...)